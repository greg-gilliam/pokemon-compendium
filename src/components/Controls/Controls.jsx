import React from "react";
import FilterForm from "../../Forms/FilterForm";

export default function Controls({ types, filterChange, selectedType }) {
  return (
    <div className="controlsContainer">
      <FilterForm
        types={types}
        filterChange={filterChange}
        selectedType={selectedType}
      />
    </div>
  );
}
