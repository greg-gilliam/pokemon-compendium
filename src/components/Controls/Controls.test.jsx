import { screen, render } from "@testing-library/react";
import Controls from "./Controls.jsx";

it("should render a dropdown with types of pokemon", async () => {
  const types = ["bug", "grass", "poison"];
  const { container } = render(
    <Controls types={types} selectedType="poison" />
  );
  const dropDown = await screen.findByText(/Filter your Pokemon!/i);
  expect(dropDown).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
