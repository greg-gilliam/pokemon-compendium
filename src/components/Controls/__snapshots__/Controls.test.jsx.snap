// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a dropdown with types of pokemon 1`] = `
<div>
  <div
    class="controlsContainer"
  >
    <div>
      <h3>
        Filter your Pokemon!
      </h3>
      <select>
        <option
          value="all"
        >
          All
        </option>
        <option />
        <option />
        <option />
      </select>
    </div>
  </div>
</div>
`;
